core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/MultiregionalBucketMigration.scala

```diff
@@ -301,30 +297,93 @@ trait MultiregionalBucketMigrationHistory extends DriverComponent with RawSqlQue
         workspace <- MonadThrow[ReadWriteAction].fromOption(workspaceOpt, NoSuchWorkspaceException(workspaceName))
 
+        maybePastBucketMigration <- getAttempt(workspace.workspaceIdAsUUID).value
+        id <- maybePastBucketMigration match {
+          case None          => scheduleFirstMigrationAttempt(workspace)
+          case Some(attempt) => restartMigrationAttempt(attempt, workspace.toWorkspaceName)
+        }
+      } yield id
+
+    private def scheduleFirstMigrationAttempt(workspace: Workspace): ReadWriteAction[Long] =
+      for {
         _ <- MonadThrow[ReadWriteAction].raiseWhen(workspace.isLocked) {
           new RawlsExceptionWithErrorReport(
-            ErrorReport(StatusCodes.BadRequest, s"'$workspaceName' bucket cannot be migrated as it is locked.")
+            ErrorReport(StatusCodes.BadRequest,
+                        s"'${workspace.toWorkspaceName}' bucket cannot be migrated as it is locked."
+            )
           )
         }
 
         _ <- MonadThrow[ReadWriteAction].raiseWhen(workspace.workspaceType == WorkspaceType.McWorkspace) {
           new RawlsExceptionWithErrorReport(
             ErrorReport(StatusCodes.BadRequest,
-                        s"'$workspaceName' bucket cannot be migrated as it is not a Google workspace."
+                        s"'${workspace.toWorkspaceName}' bucket cannot be migrated as it is not a Google workspace."
             )
           )
         }
 
-        isPending <- isPendingMigration(workspace)
-        _ <- MonadThrow[ReadWriteAction].raiseWhen(isPending) {
+        _ <- sqlu"insert into #$tableName (#$workspaceIdCol) values (${workspace.workspaceIdAsUUID})"
+        id <- sql"select LAST_INSERT_ID()".as[Long].head
+      } yield id
+
+    private def restartMigrationAttempt(pastAttempt: MultiregionalBucketMigration,
+                                        workspaceName: WorkspaceName
+    ): ReadWriteAction[Long] =
+      for {
+        _ <- MonadThrow[ReadWriteAction].raiseWhen(pastAttempt.outcome.getOrElse(Success).isSuccess) {
           new RawlsExceptionWithErrorReport(
-            ErrorReport(StatusCodes.BadRequest, s"Workspace '$workspaceName' is already pending bucket migration.")
+            ErrorReport(
+              StatusCodes.BadRequest,
+              s"Workspace '$workspaceName' ${if (pastAttempt.outcome.isDefined) "has already had its bucket migrated successfully"
+                else "is already pending bucket migration"}."
+            )
           )
         }
 
-        _ <- sqlu"insert into #$tableName (#$workspaceIdCol) values (${workspace.workspaceIdAsUUID})"
-        id <- sql"select LAST_INSERT_ID()".as[Long].head
-      } yield id
+        _ <- (pastAttempt.workspaceBucketTransferJobIssued,
+              pastAttempt.workspaceBucketTransferred,
+              pastAttempt.tmpBucketTransferJobIssued,
+              pastAttempt.tmpBucketTransferred
+        ) match {
+          case (Some(_), None, _, _) =>
```

Comment:
Can you add some comments about what these cases are doing/covering?

Reply:
The "update5" and "update3" methods aren't helping me.

Reply:
Sure, I'll add some comments to clarify!

`update5` and `update3` are methods that were ported over from PPW and describe how many columns to update at once. `update3` takes the id for a migration attempt, and then 3 column names interspersed with the value we want to set. So for the `update3` call on line 363, we're updating the finished, message, and outcome columns for `pastAttempt.id` and setting the value for all of those columns to `None`. Clearing out those values will cause the pipeline to pick the migration back up and try to migrate it again.

Reply:
Thanks!